### PR TITLE
fix: hide payment info card when empty

### PR DESCRIPTION
**Pull request by FollowTheFlo**
CXSPA-4069
Goal: handle situation when PaymentInfo has no data about credit card and/or billingAddress.

### CODE REVIEW

File: feature-libs/order/components/order-details/order-detail-billing/order-detail-billing.component.html [hunk 1]
```diff
@@ -1,15 +1,27 @@
-<div class="cx-order-items" *ngIf="order$ | async as order">
-  <div class="cx-review-summary" *ngIf="order.paymentInfo as paymentDetails">
-    <div class="cx-review-summary-card cx-review-summary-payment-card">
-      <cx-card
-        [content]="getPaymentMethodCard(paymentDetails) | async"
-      ></cx-card>
-    </div>
+<div *ngIf="true">
```
Comment (Matejk00) [inline feature-libs/order/components/order-details/order-detail-billing/order-detail-billing.component.html]: What does it check `*ngIf="true"`? Is it necessary here, maybe I am missing the point.

File: feature-libs/order/components/order-details/order-detail-billing/order-detail-billing.component.html [hunk 2]
```diff
@@ -1,15 +1,40 @@
...
+        </div>
+      </ng-container>
+      <ng-container *cxFeatureLevel="'!6.5'">
+        <div class="cx-review-summary-card cx-review-summary-payment-card">
+          <cx-card
+            [content]="getBillingAddressCard(paymentDetails) | async"
+          ></cx-card>
+        </div>
+      </ng-container>
+      <ng-container *cxFeatureLevel="'6.5'">
```
Comment (Matejk00) [inline feature-libs/order/components/order-details/order-detail-billing/order-detail-billing.component.html]: I think we reached a CF date, so all version numbers and target branch should be changed into 6.6.

**Review by Matejk00 — CHANGES_REQUESTED** [review]

File: feature-libs/order/components/order-details/order-detail-billing/order-detail-billing.component.html (hunk 1 above)
Comment (FollowTheFlo) [reply feature-libs/order/components/order-details/order-detail-billing/order-detail-billing.component.html]: My bad, forgot to remove it from testing.

Comment (FollowTheFlo) [reply feature-libs/order/components/order-details/order-detail-billing/order-detail-billing.component.html]: Looks like i forgot to push a commit, the rest of the files are not the latest version neither, reworking it.

File: feature-libs/order/styles/components/order-overview/_order-overview.scss [hunk 3]
```diff
@@ -106,6 +108,13 @@
     flex-grow: 1;
     margin-bottom: 0;
 
+    @include forVersion(6.5) {
```
Comment (Matejk00) [inline feature-libs/order/styles/components/order-overview/_order-overview.scss]: I'm wondering if this also shouldn't be 6.6 version. Just wanted to confirm :)

**Review by Matejk00 — CHANGES_REQUESTED** [review]

File: feature-libs/order/styles/components/order-overview/_order-overview.scss (hunk 3 above)
Comment (FollowTheFlo) [reply feature-libs/order/styles/components/order-overview/_order-overview.scss]: true, moved to 6.6

File: feature-libs/order/styles/components/order-overview/_order-overview.scss [hunk 4]
```diff
@@ -22,8 +22,10 @@
         flex-grow: 1;
 
         @include media-breakpoint-up(lg) {
-          border-inline-end: 1px solid var(--cx-color-text);
-          margin-inline-end: 10rem;
+          @include forVersion(1, 6.5) {
```
Comment (FollowTheFlo) [inline feature-libs/order/styles/components/order-overview/_order-overview.scss:25]: legacy code -> up to 6.5
new code below (line 111) -> from 6.6

**Review by Matejk00 — APPROVED** [review]

**Pull request merged**